--- a/net/sched/sch_cake.c
+++ b/net/sched/sch_cake.c
@@ -2542,7 +2542,7 @@
 			if (i == 2) {
 				cake_set_rate(b, q->rate_bps * 9 / 10, mtu, us_to_ns(q->target),
-					      us_to_ns(10000));
+					      us_to_ns(100000));
 			} else {
 				cake_set_rate(b, rate, mtu, us_to_ns(q->target),
 					      us_to_ns(100000));
@@ -2550,7 +2550,9 @@
 
 		if (i == 7) {
 			b->tin_quantum = 65535;
-		} else {
+		} else if (i == 2) {
+			b->tin_quantum = 65535;
+		} else {
 			b->tin_quantum = max_t(u16, 1U, quantum);
 		}
 
